camunda-bpm-camel-common/src/main/java/org/camunda/bpm/camel/component/producer/StartProcessProducer.java

```diff
@@ -76,7 +90,63 @@ public void process(Exchange exchange) throws Exception {
 
     exchange.setProperty(EXCHANGE_HEADER_PROCESS_DEFINITION_ID, instance.getProcessDefinitionId());
     exchange.setProperty(EXCHANGE_HEADER_PROCESS_INSTANCE_ID, instance.getProcessInstanceId());
-    exchange.getOut().setBody(instance.getProcessInstanceId());
+    
+    setOutBody(exchange, instance);
+    
+  }
+  
+  private void setOutBody(final Exchange exchange, final ProcessInstance instance) {
+    if (parameters.containsKey(COPY_PROCESS_VARIABLES_TO_OUT_BODY_PARAMETER)) {
+      if (historyService == null) {
```

Comment:
If the process instance is not yet ended but waits in a wait state you could also get this information from runtime. Just saying :-)

Reply:
Yes, I know. It is also mentioned in the README.

Reply:
I extended the implementation to support also runtime-only environments and accordingly adopted the README.